src/NServiceBus.Transport.RabbitMQ.CommandLine/Commands/Delays/DelaysMigrateCommand.cs

```diff
@@ -0,0 +1,215 @@
+                || message.BasicProperties == null
+                || message.BasicProperties.Headers == null
+                || !message.BasicProperties.Headers.ContainsKey(DelayInfrastructure.DelayHeader)
+                || !message.BasicProperties.Headers.ContainsKey(timeSentHeader);
+        }
+
+        public DelaysMigrateCommand(RabbitMQ.ConnectionFactory connectionFactory, IRoutingTopology routingTopology, bool quietMode, IConsole console)
+        {
+            this.connectionFactory = connectionFactory;
+            this.routingTopology = routingTopology;
+            this.quietMode = quietMode;
+            this.console = console;
+        }
+
+        public Task Run(CancellationToken cancellationToken = default)
+        {
+            using var connection = connectionFactory.CreateAdministrationConnection();
+            using var channel = connection.CreateModel();
+
+            try
+            {
+                for (int currentDelayLevel = DelayInfrastructure.MaxLevel; currentDelayLevel >= 0 && !cancellationToken.IsCancellationRequested; currentDelayLevel--)
+                {
+                    MigrateQueue(channel, currentDelayLevel, cancellationToken);
+                }
+            }
+            catch (OperationInterruptedException ex)
+            {
+                if (ex.ShutdownReason.ReplyCode == 404)
+                {
+                    console.WriteLine($"Fail: {ex.ShutdownReason.ReplyText}, run installers prior to running this tool.");
+                }
+                else
+                {
+                    console.WriteLine($"Fail: {ex.Message}");
+                }
+            }
+            finally
+            {
+                if (channel.IsOpen)
+                {
+                    channel.Close();
+                }
+
+                if (connection.IsOpen)
+                {
+                    connection.Close();
+                }
+            }
+
+            return Task.CompletedTask;
+        }
+
+        void MigrateQueue(IModel channel, int currentDelayLevel, CancellationToken cancellationToken)
+        {
+            var currentDelayQueue = $"nsb.delay-level-{currentDelayLevel:00}";
+            var messageCount = channel.MessageCount(currentDelayQueue);
+            var declaredDestinationQueues = new HashSet<string>();
+
+            if (messageCount > 0)
+            {
+                if (!quietMode)
+                {
+                    console.Write($"Processing {messageCount} messages at delay level {currentDelayLevel:00}. ");
+                }
+
+                int skippedMessages = 0;
+                int processedMessages = 0;
+
+                for (int i = 0; i < messageCount && !cancellationToken.IsCancellationRequested; i++)
+                {
+                    var message = channel.BasicGet(currentDelayQueue, false);
+
+                    if (message == null)
+                    {
+                        // Queue is empty
```

Comment:
Since we're polling for messages here instead of using consume, null would just mean the queue is empty.  This could happen if the TTL's expire while I"m looping through the messages.